File: apps/pos/native-android/app/src/main/java/com/desktopkitchen/pos/models/Modifier.kt
package com.desktopkitchen.pos.models

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class ModifierGroup(
    val id: Int,
    val name: String,
    val selection_type: String,
    val required: Boolean = false,
    val min_selections: Int = 0,
    val max_selections: Int = 0,
    val sort_order: Int = 0,
    val active: Boolean = true,
    val modifiers: List<ModifierItem>? = null
)

@JsonClass(generateAdapter = true)
data class ModifierItem(
    val id: Int,
    val group_id: Int,
    val name: String,
    val price_adjustment: Double = 0.0,
    val sort_order: Int = 0,
    val active: Boolean = true
)

File: apps/pos/native-windows/DesktopKitchenPOS/Models/Modifier.cs
using System.Text.Json.Serialization;

namespace DesktopKitchenPOS.Models;

public record ModifierGroup
{
    [JsonPropertyName("id")]
    public int Id { get; init; }

    [JsonPropertyName("name")]
    public string Name { get; init; } = "";

    [JsonPropertyName("selection_type")]
    public string SelectionType { get; init; } = "single";

    [JsonPropertyName("required")]
    public bool Required { get; init; }

    [JsonPropertyName("min_selections")]
    public int MinSelections { get; init; }

    [JsonPropertyName("max_selections")]
    public int MaxSelections { get; init; }

    [JsonPropertyName("sort_order")]
    public int SortOrder { get; init; }

    [JsonPropertyName("active")]
    public bool Active { get; init; } = true;

    [JsonPropertyName("modifiers")]
    public List<ModifierItem>? Modifiers { get; init; }
}

public record ModifierItem
{
    [JsonPropertyName("id")]
    public int Id { get; init; }

    [JsonPropertyName("group_id")]
    public int GroupId { get; init; }

    [JsonPropertyName("name")]
    public string Name { get; init; } = "";

    [JsonPropertyName("price_adjustment")]
    public double PriceAdjustment { get; init; }

    [JsonPropertyName("sort_order")]
    public int SortOrder { get; init; }

    [JsonPropertyName("active")]
    public bool Active { get; init; } = true;
}
